pkg/controller/utils/capability.go

```diff
@@ -331,7 +331,7 @@ func generateJSONSchemaWithRequiredProperty(schemas map[string]*openapi3.Schema,
 
 // GetKubeSchematicOpenAPISchema gets OpenAPI v3 schema based on kube schematic parameters for component and trait definition
 func GetKubeSchematicOpenAPISchema(params []commontypes.KubeParameter) ([]byte, error) {
-	required := []string{}
```

Comment:
Let's keep this for empty array case (not let empty array to be nil).

Reply:
ye, i will revert it